BrainPortal/app/controllers/access_profiles_controller.rb

```diff
@@ -0,0 +1,146 @@
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+# Simple controller for creating/editing access profiles, which are
+# an admin-only resource.
+class AccessProfilesController < ApplicationController
+
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+  before_filter :login_required
+  before_filter :admin_role_required
+
+  def index #:nodoc:
+
+    @scope = scope_from_session('access_profiles')
+    scope_default_order(@scope, 'name')
+
+    @base_scope       = AccessProfile.where({})
+    @access_profiles  = @scope.apply(@base_scope)
+
+    respond_to do |format|
+      format.html # index.html.erb
+    end
+  end
+
+  def show #:nodoc:
+    @access_profile = AccessProfile.find(params[:id])
+
+    respond_to do |format|
+      format.html # show.html.erb
+    end
+  end
+
+  def new #:nodoc:
+    @access_profile = AccessProfile.new
+  end
+
+  def create #:nodoc:
+    @access_profile = AccessProfile.new(params[:access_profile])
+
+    respond_to do |format|
+      if @access_profile.save
+        flash[:notice] = 'AccessProfile was successfully created.'
+        format.html  { redirect_to :action => :index }
+      else
+        format.html  { render :action  => :new }
+      end
+    end
+  end
+
+  def update #:nodoc:
+    @access_profile = AccessProfile.find(params[:id])
+
+    # Remember 'before' state for the two main associations
+    old_group_ids = @access_profile.group_ids.sort
+    old_user_ids  = @access_profile.user_ids.sort
+
+    # The success variable will be false for errors on ordinary attributes;
+    # for the group_ids and user_ids list, these are always updated with no errors or logging... :-(
+    success = @access_profile.update_attributes_with_logging(params[:access_profile], current_user)
+
+    # Adjust groups and users, and log differences
+    new_group_ids = @access_profile.group_ids.sort
+    new_user_ids  = @access_profile.user_ids.sort
+    @access_profile.addlog_object_list_updated("Projects", Group, old_group_ids, new_group_ids, current_user)
+    @access_profile.addlog_object_list_updated("Users",    User,  old_user_ids,  new_user_ids,  current_user, :login)
+
+    # Group list has changed? Apply to each affected user
+    if old_group_ids != new_group_ids
+      ap_removed_gids = old_group_ids - new_group_ids # what disappeared in the current AP
+      affected_users = User.where(:id => params[:affected_user_ids]).all
+      affected_users.each do |user|
+        orig_user_gids = user.group_ids
+        user.apply_access_profiles(remove_group_ids: ap_removed_gids) # (re)add all groups from all APs of the user, minus lost groups
+        user.addlog_object_list_updated("Updated Access Profile '#{@access_profile.name}', Projects",
+                                        Group, orig_user_gids, user.group_ids, current_user)
+      end
+    end
```

Comment:
So if:
* user A belongs to group X
* user A is added to AccessProfile P that contains groups X and Y
* X is removed from P
Then user A is removed from group X. I can see issues coming with that but I don't really have a solution :)